### PR TITLE
Too big error message #4827

### PR DESCRIPTION
**Pull request by dimmonn**
[#4827](https://github.com/JabRef/jabref/issues/4827) 

checking if the error belongs to sql state 01S00 then it gets shortened to the first dot.

ex. 
this:  **The server time zone value 'MSK' is unrecognized or represents more than one time zone. You must configure either the server or JDBC driver (via the serverTimezone configuration property) to use a more specifc time zone value if you want to utilize time zone support.**

becomes: 
**The server time zone value 'MSK' is unrecognized or represents more than one time zone**

if there is no dot in error string it remains the same length

Screenshot at May 12 09-08-29

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Thanks for your contribution. However I think it would be better if you just add the second sentence to a new line.

**Comment by dimmonn** [comment]
Thanks,
You meant to say to keep the below as the error message?

You must configure either the server or JDBC driver (via the serverTimezone configuration property) to use a more specifc time zone value if you want to utilize time zone support.

**Comment by dimmonn** [comment]
Thanks, gotcha

**Comment by tobiasdiez** [comment]
It might also work to play around with `setWidth` here:
https://github.com/JabRef/jabref/blob/3baa6bd22c1dd0fab6c3c1480fc723654afd22b2/src/main/java/org/jabref/gui/JabRefDialogService.java#L165-L178

This has the advantage, that it would fix similar errors that occur elsewhere.

**Comment by dimmonn** [comment]
Right I'll try

**Comment by dimmonn** [comment]
formatted the error message instead,  every 25th word ends up as a new line, eventually it looks like below.

(org/jabref/gui/JabRefDialogService.java:167 setWidth here works a bit unexpected with the long string lines, so I didn't play around with the setWidth)

image

File: src/main/java/org/jabref/logic/shared/DBMSConnection.java [hunk 1]
```diff
@@ -10,10 +10,10 @@
 import org.jabref.logic.shared.exception.InvalidDBMSConnectionPropertiesException;
 import org.jabref.model.database.shared.DBMSType;
 import org.jabref.model.database.shared.DatabaseConnection;
-
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+
 public class DBMSConnection implements DatabaseConnection {
```
Comment (davidemdot) [inline src/main/java/org/jabref/logic/shared/DBMSConnection.java]: Hey, @dimmonn. It looks that there is a problem with the Checkstyle test. You should insert a blank line before org.slf4j.Logger and remove one of the two empty lines after org.slf4j.LoggerFactory:

> import org.jabref.model.database.shared.DatabaseConnection;
> 
> import org.slf4j.Logger;
> import org.slf4j.LoggerFactory;
> 
> public class DBMSConnection implements DatabaseConnection {

Thanks for your contribution!

Comment (dimmonn) [reply src/main/java/org/jabref/logic/shared/DBMSConnection.java]: Got you, many thanks will fix now

File: src/test/java/org/jabref/logic/util/strings/StringFormatterTest.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+class StringFormatterTest {
+
+    @Test
+    void formatErrorMessage() {
+        StringFormatter stringFormatter = new StringFormatter();
+        String error = "The server time zone value 'MSK' is unrecognized or represents more than one time zone. You must configure either the server or JDBC driver (via the serverTimezone configuration property) to use a more specifc time zone value if you want to utilize time zone support.";
+        Exception exceptionUnderTest = new Exception(error);
+        String errorToBeThrown = stringFormatter.formatErrorMessage(exceptionUnderTest);
+        String[] lines = errorToBeThrown.split(System.lineSeparator());
```
Comment (tobiasdiez) [inline src/test/java/org/jabref/logic/util/strings/StringFormatterTest.java]: Please compare it directly to the expected error message (e.g `assertEquals("...", errorToBeThrown)).

Comment (dimmonn) [reply src/test/java/org/jabref/logic/util/strings/StringFormatterTest.java]: np

Comment (dimmonn) [reply src/test/java/org/jabref/logic/util/strings/StringFormatterTest.java]: @tobiasdiez 

any type of setWidth seems to cut off the string, unless I do it incorrectly

ex,
 exceptionDialog.getDialogPane().setMaxWidth(mainWindow.getWidth() / 3);

image

Comment (tobiasdiez) [reply src/test/java/org/jabref/logic/util/strings/StringFormatterTest.java]: Thanks for testing. It seems to be a bug in the library we use. I've proposed a fix at https://github.com/controlsfx/controlsfx/pull/1155. Let's hope they include it quickly. If not, then we can temporarily go with your fix and then at some point fix it in the dialog service class.

Comment (dimmonn) [reply src/test/java/org/jabref/logic/util/strings/StringFormatterTest.java]: Cool, thanks

Comment (tobiasdiez) [reply src/test/java/org/jabref/logic/util/strings/StringFormatterTest.java]: It should be fixed now in controlsfx. To see it you need to change 
https://github.com/JabRef/jabref/blob/master/build.gradle#L139
to `controlsfx-8.40.16`.

**Comment by dimmonn** [comment]
updated lib works well

image

not sure if some unit tests needs to cover this change

File: CHANGELOG.md [hunk 3]
```diff
@@ -114,7 +114,7 @@ We refer to [GitHub issues](https://github.com/JabRef/jabref/issues) by using `#
 - We fixed an issue where special characters where removed from non label key generation pattern parts [#4767](https://github.com/JabRef/jabref/issues/4767)
 - We fixed an issue where the RIS import would overwite the article date with the value of the acessed date [#4816](https://github.com/JabRef/jabref/issues/4816)
 - We fixed an issue where an NullPointer exception was thrown when a referenced entry in an Open/Libre Office document was no longer present in the library. Now an error message with the reference marker of the missing entry is shown. [#4932](https://github.com/JabRef/jabref/issues/4932)
-
+- We fixed an issue where the exception related to timezone was too big. [#4827](https://github.com/JabRef/jabref/issues/4827)
```
Comment (Siedlerchr) [inline CHANGELOG.md]: maybe: where a database exception related to a missing timezone was too big?
(cause otherwise it sounds a bit confusing

Comment (dimmonn) [reply CHANGELOG.md]: sure, done

**Comment by Siedlerchr** [comment]
Thanks for your contribution: PS: If you write sth like Fixes `#1234`  then the corresponding issue will be automatically closed on merge

**Pull request merged**